test/unix/test_lwt_bytes.ml

```diff
@@ -677,4 +713,92 @@ let suite = suite "lwt_bytes" [
       let check = "abcdef" = Lwt_bytes.to_string buffer in
       Lwt.return check
     end;
+
+    test "map_file" begin fun () ->
+      let test_file = "bytes_io_data" in
+      let fd = Unix.openfile test_file [O_RDONLY] 0 in
+      let shared = false in
+      let size = 6 in
+      let buffer = Lwt_bytes.map_file ~fd ~shared ~size () in
+      let check = "abcdef" = Lwt_bytes.to_string buffer in
+      let () = Unix.close fd in
+      Lwt.return check
+    end;
+
+    test "page_size" begin fun () ->
+      let size = 4096 in
+      Lwt.return (size = Lwt_bytes.page_size)
+    end;
+
+    test "mincore buffer length = page_size * 2, n_states = 1"
+      ~only_if:(fun () -> not Sys.win32) begin fun () ->
+      test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 1
+      >>= fun states ->
+      Lwt.return (states.(0) = true)
+    end;
+
+    test "mincore buffer length = page_size * 2, n_states = 2"
+      ~only_if:(fun () -> not Sys.win32) begin fun () ->
+      Lwt.catch
+        (fun () ->
+        test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 2
+        >>= fun states ->
+        Lwt.return (states.(0) = false)
```

Comment:
This test is still capable of passing even if the exception is not raised, if `states.(0)` turns out to be `false`. I think it needs to unconditionally do `Lwt.return false` on this line. Compare with the `wait_mincore` tests that expect an exception.